airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/source.py

```diff
@@ -23,38 +23,136 @@
 """
 
 import json
-import os
+import sys
+from typing import Dict, List
 
-from airbyte_protocol import AirbyteCatalog, AirbyteConnectionStatus, Status
-from base_python import AirbyteLogger, CatalogHelper
-from base_singer import SingerHelper, SingerSource
+from airbyte_protocol import AirbyteConnectionStatus, Status
+from base_python import AirbyteLogger
+from base_singer import SingerSource, SyncMode, SyncModeInfo
+from googleads import adwords, oauth2
+from tap_adwords import VERSION
 
 
 class SourceGoogleAdwordsSinger(SingerSource):
```

Comment:
You can use BaseSingerSourse, this will slightly reduce the amount of code, for example, it will remove the disсover_сmd and the check_config functions, WDYT?